pkg/apiserver/logging/logging.go

```diff
@@ -188,22 +139,86 @@ func LoggingDeleteFluentbitOutput(request *restful.Request, response *restful.Re
 	response.WriteAsJson(res)
 }
 
-func logQuery(level log.LogQueryLevel, request *restful.Request) (*v1alpha2.QueryResult, error) {
+func logQuery(param v1alpha2.QueryParameters, response *restful.Response) {
+	es, err := cs.ClientSets().ElasticSearch()
+	if err != nil {
+		response.WriteHeaderAndEntity(http.StatusServiceUnavailable, errors.Wrap(err))
+		return
+	}
+
+	res, err := es.Query(param)
+	if err != nil {
+		response.WriteHeaderAndEntity(http.StatusInternalServerError, errors.Wrap(err))
+		return
+	}
+
+	response.WriteAsJson(res)
+}
+
+func logExport(param v1alpha2.QueryParameters, response *restful.Response) {
 	es, err := cs.ClientSets().ElasticSearch()
 	if err != nil {
-		klog.Error(err)
-		return nil, err
+		response.WriteHeaderAndEntity(http.StatusServiceUnavailable, errors.Wrap(err))
+		return
+	}
+
+	response.Header().Set("Content-Type", restful.MIME_OCTET)
+
+	// keep search context alive for 1m
+	param.ScrollTimeout = time.Minute
+	// export 1000 records in every iteration
+	param.Size = 1000
+	// from is not allowed in a scroll context
+	param.From = 0
+
+	var scrollId string
+	// limit to retrieve max 100k records
+	for i := 0; i < 100; i++ {
+		var res *v1alpha2.QueryResult
+		if scrollId == "" {
```

Comment:
What if user cancelled request in the middle of downloading? I believe this loop will continue. Need to break loop quickly if user cancelled downloading. You can try this https://gist.github.com/zryfish/353dddb2f7eb4dd99e195b0270d52e9b

Reply:
we interrupt downloading by capture error from io.Copy

Reply:
no, that's not right. Even request is cancelled, response is still there, keep writing. I modified the example, you can try